### PR TITLE
Codecs (part 6) - ABI definitions

### PR DESCRIPTION
**Pull request by andreibancioiu**
This is part of a series of pull requests.

See: https://github.com/multiversx/mx-sdk-py/pull/32.

Defines ABI definition concepts, as Python classes.

### CODE REVIEW

**Review by popenta — DISMISSED** [review]

File: multiversx_sdk/abi/abi_definition.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    @classmethod
+    def load(cls, path: Path) -> "AbiDefinition":
+        content = Path(path).read_text()
+        data = json.loads(content)
+        return cls.from_dict(data)
+
+
+class EndpointDefinition:
+    def __init__(self,
+                 name: str,
```
Comment (axenteoctavian) [inline multiversx_sdk/abi/abi_definition.py:56]: Maybe we can also have `docs` here

File: multiversx_sdk/abi/abi_definition.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+        )
+
+    def __repr__(self):
+        return f"EndpointDefinition(name={self.name})"
+
+
+class ParameterDefinition:
+    def __init__(self, name: str, type: str) -> None:
+        self.name = name
+        self.type = type
```
Comment (axenteoctavian) [inline multiversx_sdk/abi/abi_definition.py:90]: Maybe (I don't know if those are relevant) `multi_arg: bool` for inputs or `multi_result: bool` for outputs

File: multiversx_sdk/abi/abi_definition.py (hunk 1 above)
Comment (andreibancioiu) [reply multiversx_sdk/abi/abi_definition.py:56]: Added.

File: multiversx_sdk/abi/abi_definition.py (hunk 2 above)
Comment (andreibancioiu) [reply multiversx_sdk/abi/abi_definition.py:93]: They are redundant / not relevant, indeed. Sufficient information is captured in the type expression itself.

**Review by popenta — APPROVED** [review]

**Review by axenteoctavian — APPROVED** [review]

**Pull request merged**